### DISCUSSION TITLE
CC1101 version 0x17 ?

### DISCUSSION THREAD

**Discussion by asafteirobert** [jgromes/RadioLib#349] [General]
I was having issues initializing a CC1101. After enabling debug mode, it seems `CC1101::getChipVersion()` always returns 0x17, but the library will only work if version is 0x14 or 0x04.
After changing `CC1101_VERSION_CURRENT` to 0x17, it initialized and started transmitting correctly.

The datasheet does mention that `VERSION` is "Subject to change without notice.", so maybe checking for hard-coded version numbers in `CC1101::begin` is not a very good idea.

Has anyone encountered such a chip? It is likely a Chinese clone so anything is to be expected.

**Comment by jgromes** [comment]
I haven't seen 0x17 used as version for CC1101, we can add it to the "whitelist".

> maybe checking for hard-coded version numbers in CC1101::begin is not a very good idea.

Version number check is used for two things:
1. To verify communication with the device is working correctly
2. To verify you're actually communicating with the correct device.

The version number is usually left unchanged throughout the chip lifetime, since the manufacturers often use it to mark different products or variants of the product - assuming of course it's still the same manufacturer. As far as I can tell, only CC1101 exists with mupltile different version numbers (see https://github.com/jgromes/RadioLib/issues/183), all other modules supported by RadioLib only have a single known variant.

**Comment by adminy** [comment]
version 14 is okay yes?
